test/common/http/matching/inputs_test.cc

```diff
@@ -107,6 +107,29 @@ TEST(MatchingData, HttpResponseTrailersDataInput) {
   }
 }
 
+TEST(MatchingData, HttpRequestQueryParamsDataInput) {
+  Network::ConnectionInfoSetterImpl connection_info_provider(
+      std::make_shared<Network::Address::Ipv4Instance>(80),
+      std::make_shared<Network::Address::Ipv4Instance>(80));
+  HttpMatchingDataImpl data(connection_info_provider);
+
+  {
+    HttpRequestQueryParamsDataInput input("username", false /* url_decoded */);
+    TestRequestHeaderMapImpl request_headers({{":path", "/test?username=foo&username=bar"}});
+    data.onRequestHeaders(request_headers);
+
+    EXPECT_EQ(input.get(data).data_, "foo");
+  }
+
+  {
+    HttpRequestQueryParamsDataInput input("user name", true /* url_decoded */);
+    TestRequestHeaderMapImpl request_headers({{":path", "/test?user\%20name=foo\%20bar"}});
+    data.onRequestHeaders(request_headers);
+
+    EXPECT_EQ(input.get(data).data_, "foo bar");
+  }
+}
+
```

Comment:
Could add some negative test cases? For example the return status is `return {AllDataAvailable, absl::nullopt}`.

Once you fixes the CI/Docs issue, the full CD/CI can run and it may ask for test coverage for the negative coverage

Reply:
Added here: https://github.com/envoyproxy/envoy/pull/24627/commits/5305a220ce918ae6fbac237d031bcb97fed2ff13#diff-53f65a7ea6acbe833684b356ca5108be6eda4de221cd37269ac94f823f9f9662R132